src/RatcliffDDM.jl

```diff
@@ -1,371 +1,662 @@
+        else
+            S += 2 * y
+        end
+
+    end
+    
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / ((ub_t - lb_t) + (ub_z - lb_z))  # the right function if pdf_sv()/sz or pdf_sv()/st
+    
+    return (ht + hz) * S / 3
+
+end
+
+# Simpson's Method two dimentional case
+function _simpson_2D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
+ #   @assert (n_sz & 1) == 0 && (n_st & 1) == 0     # n_st and n_sz have to be even
+ #   @assert (ub_t-lb_t)*(ub_z-lb_z)>0 && (n_sz*n_st)>0  # 2D-integration only
+
+    ht = (ub_t-lb_t)/n_st
+    S = _simpson_1D(x, ν, η, α, z, τ, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
+
+    t_tag = 0
+    y = 0 
+    for i_t in 1:n_st
+        t_tag = lb_t + ht * i_t
+        y = _simpson_1D(x, ν, η, α, z, t_tag, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
+
+        if isodd(i_t)
+            S += 4 * y
+        else
+            S += 2 * y
+        end
+    end
+
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / (ub_t - lb_t)
+
+    return ht * S / 3
+
+end
+
+logpdf(d::RatcliffDDM, choice, rt; ϵ::Real = 1.0e-12) = log(pdf(d, choice, rt; ϵ))
+
+function logpdf(d::RatcliffDDM, data::T) where {T<:NamedTuple}
+    return sum(logpdf.(d, data...))
+end
+
+function logpdf(dist::RatcliffDDM, data::Array{<:Tuple,1})
+    LL = 0.0
+    for d in data
+        LL += logpdf(dist, d...)
+    end
+    return LL
+end
+
+logpdf(d::RatcliffDDM, data::Tuple) = logpdf(d, data...)
+
+########################################################################################################################################################################
+# Calculate Drift-diffusion Probability Density
+#
+# compute the densities g- and g+ of the first exit time. `cdf` implement A1 to A4 equations in Voss, Rothermund, and Voss (2004). These equations calculate Ratcliff's
+# drift-diffusion model (1978). This source codes are derived from Henrik Singmann's Density.h (rtdists) &  Voss & Voss's density.c (fast-dm).
+#
+#  * ------------------------------------------------------------------------
+#  * A verbatim copy of Jochen Voss & Andreas Voss's copyright.
+#  * ------------------------------------------------------------------------
+#  * Copyright (C) 2012  Andreas Voss, Jochen Voss.
+#  *
+#  * This program is free software; you can redistribute it and/or
+#  * modify it under the terms of the GNU General Public License as
+#  * published by the Free Software Foundation; either version 2 of the
+#  * License, or (at your option) any later version.
+#  *
+#  * This program is distributed in the hope that it will be useful, but
+#  * WITHOUT ANY WARRANTY; without even the implied warranty of
+#  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  * General Public License for more details.
+#  *
+#  * You should have received a copy of the GNU General Public License
```

Comment:
As I mentioned during our initial conversation, the GNU license is a problem because it is viral unlike the MIT licence. So if we add the GNU license to this repo, it will become GNU, and so will any code that uses this package.  I recommend reading through any papers that their code is based on and using that as a guide. Basically, it shouldn't look like you did a verbatim translation of their code into Julia. In some cases, there might be some degree of similarity because there is only one way to implement certain ideas. Does that make sense?

The other option would be to do a direct translation with the same naming conventions and put the model in its own package which could extend the same interface defined here.

Reply:
> The other option would be to do a direct translation with the same naming conventions and put the model in its own package which could extend the same interface defined here.

I think paraphrasing the code to avoid copyright issues and keeping everything in one package would be the best for users

Reply:
Thanks! Sorry. I realize it's a pain. There are somethings about  GNU (GPL) that I like better than MIT, but the main drawback with GNU that many people will not use your code if it is under GNU.  I can't say I fully understand the minutia, but it seems like it could cause issues.